test/test_rpc_endpoints.py

```diff
@@ -0,0 +1,803 @@
+            "requested_scope": "FULL_TXN_AND_RECEIPTS"
+        }
+    )
+    block = resp["result"]
+
+    assert block["transactions"] == [{
+        "txn_hash": transaction_hash,
+        "max_fee": "0x0",
+        "contract_address": contract_address,
+        "calldata": None,
+        "entry_point_selector": None,
+        "actual_fee": "0x0",
+        "status": "ACCEPTED_ON_L2",
+        "statusData": "",
+        "messages_sent": [],
+        "l1_origin_message": None,
+        "events": []
+    }]
+
+
+def test_get_block_by_hash_raises_on_incorrect_hash(deploy_info):
+    """
+    Get block by incorrect hash
+    """
+    ex = rpc_call(
+        "starknet_getBlockByHash", params={"block_hash": "0x0"}
+    )
+
+    assert ex["error"] == {
+        "code": 24,
+        "message": "Invalid block hash"
+    }
+
+
+@pytest.mark.xfail
+def test_get_state_update_by_hash(deploy_info, invoke_info):
+    """
+    Get state update for the block
+    """
+    block_with_deploy = gateway_call("get_block", blockNumber=0)
+    block_with_invoke = gateway_call("get_block", blockNumber=1)
+
+    contract_address: str = deploy_info["address"]
+    block_with_deploy_hash: str = block_with_deploy["block_hash"]
+    block_with_invoke_hash: str = block_with_invoke["block_hash"]
+
+    resp = rpc_call(
+        "starknet_getStateUpdateByHash", params={
+            "block_hash": block_with_deploy_hash
+        }
+    )
+    state_update = resp["result"]
+
+    assert state_update["block_hash"] == block_with_deploy_hash
+    assert "new_root" in state_update
+    assert isinstance(state_update["new_root"], str)
+    assert "old_root" in state_update
+    assert isinstance(state_update["old_root"], str)
+    assert "accepted_time" in state_update
+    assert isinstance(state_update["accepted_time"], int)
+    assert state_update["state_diff"] == {
+        "storage_diffs": [],
+        "contracts": [
+            {
+                "address": contract_address,
+                "contract_hash": "06f8d704f5a8f6bcb53a1c87c6f8d466ab0aaa5cf962084ac03a2145aac2d823",  # FIXME this is hardcoded but I don't know how this contract
+            }                                                                                         #     hash is calculated otherwise
+        ],
+    }
+
+    resp = rpc_call(
+        "starknet_getStateUpdateByHash", params={
+            "block_hash": block_with_invoke_hash
+        }
+    )
+    state_update = resp["result"]
+
+    assert state_update["block_hash"] == block_with_invoke_hash
+    assert "new_root" in state_update
+    assert isinstance(state_update["new_root"], str)
+    assert "old_root" in state_update
+    assert isinstance(state_update["old_root"], str)
+    assert "accepted_time" in state_update
+    assert isinstance(state_update["accepted_time"], int)
+    assert state_update["state_diff"] == {
+        "storage_diffs": [
+            {
+                # FIXME this will fail because of bug in devnet
```

Comment:
I think storage diffs should show that there was a change when balance increased from 0 to 10, but it doesn't. I will be opening a proper issue soon.

Reply:
This will be updated